### ISSUE TITLE
Add an R interface i.e. export data to R data object

### ISSUE THREAD

**Issue by nichtleiter** [sdaps/sdaps#116]
This issue is related to: #4  

For those who want to analyse the scanned survey data with R the current export functionality of sdaps is not ideal. It takes some effort until the data from the exported csv file is processed  and can be evaluated. Especially the proper naming of the columns and the factorization of the values is rather clumsy.

I already thought about writing a script that reads the `questionnaire.sdaps` file to set the column names and factors of the exported csv file to increase it's readability and usability. However this seems to be overkill since the data can already be readily accessed with the python interface. Unfortunately i don't have any experience with python so i haven't managed to access the data in it's native environment yet.

Maybe it is possible for the author or some experienced sdaps enthusiasts to look into libraries like `feather` which supports to export python objects into R data frames.

I think this feature would make R users very happy!

Thanks for reading 👍 

https://github.com/wesm/feather

**Comment by benzea** [comment]
As I don't really know anything about R. It would probably be helpful if you could describe a data format that would be easy (and efficient) to work with.

I suppose that R is array based, so as a starting point, do we have an array of sheets (questionnaires) or an array per answer across all sheets (i.e. rows or columns first compared to the CSV).

**Comment by benzea** [comment]
Note that "easy" might just mean easy to convert into something usable as the first step.

**Comment by nichtleiter** [comment]
The preferred data format in R for observational data is a data frame, just like a two dimensional array. A row represents an observation and a column represents a variable. 

Applied to survey data each questionnaire should be placed in its own row, each item should be placed in its own column and each cell corresponds to one answer. This is basically the rectangular data structure of a csv file.

The issue i have with sdaps' csv files is, that the whole naming scheme of the questionnaire is lost. For example if i look into a csv file the name of each column corresponds to the number part saved in `QObject-Head`. The textual information of `QObject-Head` ist lost. 

In addition the answers for each item are saved as integer. The information which integer corresponds to which `Answer-Mark` is also lost.

So before i can start to explore the data i have to adjust the name of each column and must replace the integers with the corresponding answer mark. This is kind of annoying because sdaps already has the information and still you have to update the data by hand.

**Comment by benzea** [comment]
Ok, thanks for the info.

You might be aware yet, but in the unstable branch you can actually specify the column headers in the LaTeX file. That still won't give you the upper/lower limit of ranges, but it should solve part of the issue at least.

**Comment by nichtleiter** [comment]
I wasn't aware about this feature. It would be great if you tell me where i can find more information about how to get a more complete csv file?

Does the master branch equals the unstable branch?

**Comment by benzea** [comment]
Yes, unstable releases come from the master branch. Have a look at https://sdaps.org/class-doc

The changes are mostly on the LaTeX side with some work to support it all in python.

**Comment by benzea** [comment]
Hehe, I just realised that a simple:

python -c "import feather; import pandas; feather.write_dataframe(pandas.read_csv('data_1.csv'), 'blub.feather')"

will already do the trick. Obviously a native export will be better as it can have more accurate data types. Now I wonder if adding a `pandas` dependency is feasible, as I then could get rid of some of the CVS related code.

**Comment by nichtleiter** [comment]
Thanks for pointing out `sdaps-class` i will look into it for my next surveys.

Related to your second answer:

Although i proposed it i don't have easy access to `feather` on my debian testing machine at the moment (`feather` is currently only in unstable). However i don't think that your proposal solves my original problem. I guess that your commands only convert the csv file into a data-frame object without renaming the columns.

I definitely vouch for a native export feature, that would make things much morge comfortable in the future.

As always thanks for your quick replies and your relentless effort to improve sdaps!

**Comment by benzea** [comment]
So, I merged a feather exporter already (I seem to have forgotten about this issue). But, you mention column names, and that situation has not changed.

So, in the current design of SDAPS, a specific variable is assigned to each question. You can set this variable using the `var=` optional keyword argument and that way you can choose the values of the headers.

I do see the point of just using the human readable string from the metadata (that SDAPS has already anyway). I don't feel it is technically superior, but I can see how it is useful in a lot of scenarios. We really could make this an option for the exporters, but I am not sure about the exact scheme. I think I would somewhat prefer it to be a separate metadata export feature that you can then map onto the column names in R.

(I'll leave the bug open for a bit, but will probably simply close it soon; best thing is to open a new issue specifically on column names and/or metadata export.)
